clang/test/Driver/sycl-device-lib.cpp

```diff
@@ -21,6 +21,7 @@
 // SYCL_DEVICE_LIB_UNBUNDLE_DEFAULT-NEXT: clang-offload-bundler{{.*}} "-type=o" "-targets=sycl-spir64-unknown-unknown-sycldevice" "-inputs={{.*}}libsycl-cmath.o" "-outputs={{.*}}libsycl-cmath-{{.*}}.o" "-unbundle"
 // SYCL_DEVICE_LIB_UNBUNDLE_DEFAULT-NEXT: clang-offload-bundler{{.*}} "-type=o" "-targets=sycl-spir64-unknown-unknown-sycldevice" "-inputs={{.*}}libsycl-cmath-fp64.o" "-outputs={{.*}}libsycl-cmath-fp64-{{.*}}.o" "-unbundle"
 // SYCL_DEVICE_LIB_UNBUNDLE_DEFAULT-NEXT: clang-offload-bundler{{.*}} "-type=o" "-targets=sycl-spir64-unknown-unknown-sycldevice" "-inputs={{.*}}libsycl-fallback-cassert.o" "-outputs={{.*}}libsycl-fallback-cassert-{{.*}}.o" "-unbundle"
+// SYCL_DEVICE_LIB_UNBUNDLE_DEFAULT-NEXT: clang-offload-bundler{{.*}} "-type=o" "-targets=sycl-spir64-unknown-unknown-sycldevice" "-inputs={{.*}}libsycl-fallback-cstring.o" "-outputs={{.*}}libsycl-fallback-cstring-{{.*}}.o" "-unbundle"
```

Comment:
Please, add clang/test/Driver/Inputs/SYCL/lib/libsycl-fallback-cstring.o. The fact that the test is passing w/o this file proves that https://github.com/intel/llvm/issues/3630 is not fixed. I think `--sysroot` just **adds** new search paths, whereas we need to **replace** default paths with the new location.

Reply:
Hi, @bader 
Currently, we follow CUDA( https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/Cuda.cpp#L168 ), sysroot will be added as a candidate and we will go through all candidates to find existing libdevice binary. If we added fake binary for all devicelib, the clang lit test should pass even if we didn't buid libdevice.

Reply:
There is a difference between SYCL and CUDA. SYCL toolchain always adds driver directory as a candidate, whereas CUDA configures candidates via `sysroot` only.

Reply:
I suggest we change [this line](https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/SYCL.cpp#L27): from
```c++
InstallationCandidates.emplace_back(D.Dir + "/..");
```
to
```c++
InstallationCandidates.emplace_back(D.SysRoot + D.Dir + "/..");
```

Reply:
Hi, @bader 
Currently, the D.dir is an absolute path and is it correct for us to concatenate D.SysRoot with an absolute path?
Hi, @mdtoguchi , do you have any comments?
Thanks very much.

Reply:
> Currently, the D.dir is an absolute path and is it correct for us to concatenate D.SysRoot with an absolute path?

Yes.

Reply:
Marking this conversation as resolved:
- `clang/test/Driver/Inputs/SYCL/lib/libsycl-fallback-cstring.o` is added
- https://github.com/intel/llvm/issues/3630 can be handled by another PR.